internal/character/pela/trace.go

```diff
@@ -0,0 +1,84 @@
+package pela
+
+import (
+	"github.com/simimpact/srsim/pkg/engine/event"
+	"github.com/simimpact/srsim/pkg/engine/info"
+	"github.com/simimpact/srsim/pkg/engine/modifier"
+	"github.com/simimpact/srsim/pkg/engine/prop"
+	"github.com/simimpact/srsim/pkg/key"
+	"github.com/simimpact/srsim/pkg/model"
+)
+
+// A2:
+// 	Deals 20% more DMG to debuffed enemies.
+// A4:
+//	When Pela is on the battlefield, all allies' Effect Hit Rate increases by 10%.
+// A6:
+//	Using Skill to remove buff(s) increases the DMG of the next attack by 20%.
+
+const (
+	A2 key.Modifier = "pela-a2"
+	A4 key.Modifier = "pela-a4"
+	A6 key.Modifier = "pela-a6"
+)
+
+func init() {
+	modifier.Register(A2, modifier.Config{
+		Listeners: modifier.Listeners{
+			OnBeforeHitAll: func(mod *modifier.ModifierInstance, e event.HitStartEvent) {
+				dmgBoost := 0.0
+				if mod.Engine().ModifierCount(e.Hit.Defender.ID(), model.StatusType_STATUS_DEBUFF) >= 1 {
+					dmgBoost = 0.2
+				}
+				mod.SetProperty(prop.AllDamagePercent, dmgBoost)
+			},
```

Comment:
In all `OnBeforeHit*` listeners, you need to modify the `Hit` data in the event rather than the modifier itself. `HitStartEvent` contains a snapshot of the `Attacker` and `Defender` stats (`info.Stats`) which is what is used for the hit calculation. Changes to the modifier would only apply to the next hit (and persist beyond this hit call):
```go
if mod.Engine().ModifierCount(e.Hit.Defender.ID(), model.StatusType_STATUS_DEBUFF) >= 1 {
  e.Hit.Attacker.AddProperty(prop.AllDamagePercent, 0.2)
}
```